ngraph/core/src/op/einsum.cpp

```diff
@@ -0,0 +1,299 @@
+                        pshape.begin() + dim_ind, pshape.begin() + dim_ind + num_broadcasted_dims));
+                    if (label_to_shape.find(label) == label_to_shape.end())
+                    {
+                        label_to_shape[label] = current_sub_pshape;
+                    }
+                    else
+                    {
+                        bool is_broadcast_success =
+                            PartialShape::broadcast_merge_into(label_to_shape[label],
+                                                               current_sub_pshape,
+                                                               op::AutoBroadcastType::NUMPY);
+                        NODE_VALIDATION_CHECK(this,
+                                              is_broadcast_success == true,
+                                              "Input dimensions labeled with ellipsis for Einsum "
+                                              "must be broadcastable.");
+                    }
+                    dim_ind += num_broadcasted_dims;
+                }
+                else
+                {
+                    if (label_to_shape.find(label) == label_to_shape.end())
+                    {
+                        label_to_shape[label] = PartialShape{pshape[dim_ind]};
+                    }
+                    else
+                    {
+                        NODE_VALIDATION_CHECK(
+                            this,
+                            label_to_shape[label].compatible(PartialShape{pshape[label_ind]}),
+                            "Different input dimensions indicated by the same labels for Einsum "
+                            "must be compatible.");
+                        PartialShape::merge_into(label_to_shape[label],
+                                                 PartialShape{pshape[dim_ind]});
+                    }
+                    ++dim_ind;
+                }
+            }
+        }
+        else
+        {
+            for (auto const& label : labels)
+            {
+                NODE_VALIDATION_CHECK(this,
+                                      label != "...",
+                                      "The subscript corresponding to a dynamic rank input must "
+                                      "not contain ellipsis.");
+
+                if (label_to_shape.find(label) == label_to_shape.end())
+                {
+                    label_to_shape[label] = PartialShape{Dimension::dynamic()};
+                }
+            }
+        }
+    }
+
+    // compute the output shape
+    std::vector<std::string> output_labels;
+    output_labels = extract_labels(output_subscript);
+    std::vector<Dimension> output_pshape_vector;
+
+    for (auto const& output_label : output_labels)
+    {
+        NODE_VALIDATION_CHECK(this,
+                              label_to_shape.find(output_label) != label_to_shape.end(),
+                              "Label in output subscript of Einsum equation must enter at least "
```

Comment:
nit: word "enter" is not appropriate here. Do you mean "introduce"?

Reply:
Enter means present in this case.